pkg/controller/handler.go

```diff
@@ -201,8 +217,24 @@ func (p *Propeller) Handle(ctx context.Context, namespace, name string) error {
 	}
 
 	for streak = 0; streak < maxLength; streak++ {
+		// load the static blob every time the FlyteWorkflow is getting processed
+		// because the workflow is rewritten after the update at the end of the loop
+		if len(w.WorkflowClosureDataReference) != 0 {
+			err := k8s.MergeWorkflowClosure(w, workflowClosure)
```

Comment:
I don't love this function. If we are taking the workflowClosureStore retrieve out of this loop we might as well parse the closure there as well. I think the advantages are twofold (1) we don't need to parse the workflow closure at every iteration, incurring additional unnecesary overrhead (if it is very large this may be non-negligible) and (2) encapsulating the workflow update in this function makes this more difficult to follow.